### DISCUSSION TITLE
Object console.warn working, but will not add data to Firebase

### DISCUSSION THREAD

**Discussion by thedigitalmenagerie** [nss-evening-cohort-14/evening-client-side#301] [Capstones!] [answered]
Honey-Rae

I am not able to add the friend or UserFriend object in the handClick() at UserCardComponent. I am able to console warn the objects and their corresponding key value pairs. I am thinking maybe there is an issue in my logic to add a user as friend or in the Firebase calls (due to error message) to add the friend and userFriend data.

(X2)
vendors~main.chunk.js:75080 POST https://hr-captstone-gratitude-default-rtdb.firebaseio.com//friends.json 400 (Bad Request)
vendors~main.chunk.js:75496 Uncaught (in promise) Error: Request failed with status code 400

What have you googled?
https://www.pluralsight.com/guides/add-data-into-an-array-in-a-state-object
https://stackoverflow.com/questions/19671317/400-bad-request-http-error-code-meaning
https://www.digitalocean.com/community/tutorials/how-to-call-web-apis-with-the-useeffect-hook-in-react

What have you tried? 
I tried using a useEffect() and useState(). One within another (lots of errors).
I tried adding dummy data to FB. (I thought maybe it didn't recognize the data due to being empty)
I tried several instances of reworking logic and the furthest I could get in each iteration was to console.warn the objects. All attempts got me to the same error. 

https://github.com/thedigitalmenagerie/hr-gratitude-capstone/blob/userFriends/src/Components/UserCardComponent.js

**Comment by ajawashington** [comment, marked as answer]
@thedigitalmenagerie you are passing a Friend obj when you call addFriend on your userCardComponent, but you are not including that in your Axios request, and attaching it to the request. 

https://github.com/thedigitalmenagerie/hr-gratitude-capstone/blob/7bfd58fa289969563333e68b6552bc90a8d808f1/src/helpers/data/FriendData.js#L18

**Reply by thedigitalmenagerie** [reply]
I did this right before I checked here. Now, it completes post but will not complete patch and doesn't add to Firebase.

**Reply by thedigitalmenagerie** [reply]
In addition to what was mentioned above, I was trying to patch without body being passed or defined in my firebase call for add userFriend. This was nesting my data after fixing the first error.
